lib/seahorn/BvOpSem2.cc

```diff
@@ -835,6 +835,44 @@ class OpSemVisitor : public InstVisitor<OpSemVisitor>, OpSemVisitorBase {
         m_ctx.setInstruction(*me);
       }
     } break;
+    case Intrinsic::sadd_with_overflow: {
+      // TODO: Does not handle carry bit yet
+      Type *ty = I.getOperand(0)->getType();
+      Expr op0 = lookup(*I.getOperand(0));
+      Expr op1 = lookup(*I.getOperand(1));
+      if (ty->isVectorTy()) {
+        llvm_unreachable(nullptr);
+      }
+      Expr add_res = m_ctx.alu().doAdd(op0, op1, ty->getScalarSizeInBits());
+      // pad result with zeros to match sadd return record width
+      // assume doAdd has same bitwidth as Op0
+
+      Expr overflow =
+          m_ctx.alu().doSaddNoOverflow(op0, op1, ty->getScalarSizeInBits());
+      Expr underflow =
+          m_ctx.alu().doBaddNoUnderflow(op0, op1, ty->getScalarSizeInBits());
+      Expr is_carry = m_ctx.alu().doBNeg(
+          m_ctx.alu().doAnd(overflow, underflow, 1 /* bitwidth */),
+          1 /* bitwidth*/);
+      Expr carry_1bv = m_ctx.alu().boolToBv1(is_carry);
+
+      const DataLayout DL = m_sem.getDataLayout();
+      Type *curTy = I.getType();
+      auto *STy = dyn_cast<StructType>(curTy);
+      const StructLayout *SL = DL.getStructLayout(STy);
+      auto struc_size_in_bits = SL->getSizeInBits();
+      auto offset_in_bits = SL->getElementOffsetInBits(1);
+      auto carry_size_in_bits = (STy->getElementType(1))->getScalarSizeInBits();
+      LOG("opsem", WARN << "Struct field1 size (in bits):" << struc_size_in_bits
+                        << " " << offset_in_bits << " " << carry_size_in_bits
+                        << "\n");
+      Expr carry =
+          bv::concat(carry_1bv, m_ctx.alu().si(0U, struc_size_in_bits - offset_in_bits -
```

Comment:
switch arguments, `carry_1bv` must be in the least significant bit, not most significant as it is now